packages/integrations/react/vnode-children.js

```diff
@@ -16,7 +16,7 @@ export default function convert(children) {
 		} else if (node.type === ELEMENT_NODE) {
 			const { class: className, ...props } = node.attributes;
 			// NOTE: do not manually pass `children`, React handles this internally
-			newNode = createElement(node.name, { ...props, className, key: `${id}-${key++}` });
+			newNode = createElement(node.name, { ...props, className, key: `${id}-${key++}` }, []);
```

Comment:
It broke when I upgraded an example app, I think React stopped adding the `children` property if there were no children. Passing an empty array made it so that `vnode.props.children` was always an array.

I think a more "correct" solution would be to not call `createElement` here, but instead to create our own little vnode tree, and then at the end of walk to turn it into a React vnode.

Reply:
Ah that is annoying, but would make sense... I guess we do know if the node has children at this point and pre-allocate an array with the correct length? Or avoid creating one if there aren't any child nodes.

Reply:
Hm, I'm not sure what you're suggesting. To rephrase, ideally we should call `createElement` with the full element including all of its children and not mutate the returned vnode, because it is not intended to be mutated and could change shape between React releases (as happened here). 

If you look at the client code in this PR, it does this by using recursion to create the full React element.

Reply:
Yeah I think refactoring to use recursion here would be wise—you could get rid of the `walkSync` and probably wouldn't need the `WeakMap` to convert between node types either.

Reply:
Updated with it reimplemented as recursion